provisioning/resources/control-plane/snowplow-mini-control-plane-api.go

```diff
@@ -20,34 +20,248 @@
 package main
 
 import (
-    "github.com/emicklei/go-restful"
-    "io"
-    "net/http"
-    "log"
-    "os/exec"
-    "flag"
+  "io"
+  "net/http"
+  "log"
+  "os/exec"
+  "os"
+  "flag"
+  "io/ioutil"
 )
 
-//global variable for script's path
+// script file names
+var restartServicesScript = "restart_SP_services.sh"
+var addExternalIgluServerScript = "add_external_iglu_server.sh"
+var addIgluSuperUUIDScript = "add_iglu_server_super_uuid.sh"
+var changeUsernameAndPasswordScript = "submit_username_password_for_basic_auth.sh"
+var addDomainNameScript = "write_domain_name_to_caddyfile.sh"
+
+// global variables for paths from flags
 var scriptsPath string
+var enrichmentsPath string
+var configPath string
+var spMiniVersionFile string
 
 func main() {
-    scriptsPathFlag := flag.String("scriptsPath", "", "path for control-plane-api scripts")
-    flag.Parse()
-    scriptsPath = *scriptsPathFlag
+  scriptsPathFlag := flag.String("scriptsPath", "", "path for control-plane-api scripts")
+  enrichmentsPathFlag := flag.String("enrichmentsPath", "", "path for enrichment files")
+  configPathFlag := flag.String("configPath", "", "path for config files")
+  spMiniVersionFlag := flag.String("spMiniVersionFile", "", "path for Snowplow Mini version file")
+  flag.Parse()
+  scriptsPath = *scriptsPathFlag
+  enrichmentsPath = *enrichmentsPathFlag
+  configPath = *configPathFlag
+  spMiniVersionFile = *spMiniVersionFlag
+
+  http.HandleFunc("/restart-services", restartSPServices)
+  http.HandleFunc("/upload-enrichments", uploadEnrichments)
+  http.HandleFunc("/add-external-iglu-server", addExternalIgluServer)
+  http.HandleFunc("/add-iglu-server-super-uuid", addIgluServerSuperUUID)
+  http.HandleFunc("/change-username-and-password", changeUsernameAndPassword)
+  http.HandleFunc("/add-domain-name", addDomainName)
+  http.HandleFunc("/version", getSpminiVersion)
```

Comment:
Not sure I like all of these API endpoints... could you have a read of this (https://hackernoon.com/restful-api-designing-guidelines-the-best-practices-60e1d954e7c9) and see if you can come up with some cleaner names for these.

@alexanderdean what are your thoughts here?